src/sentry/static/sentry/app/views/stream.jsx

```diff
@@ -113,7 +113,10 @@ const Stream = createReactClass({
       return;
     }
 
-    this.fetchData();
+    // Do not make new API request if props haven't actually changed
+    if (!_.isEqual(this.props, nextProps)) {
+      this.fetchData();
+    }
```

Comment:
since the other stuff in this method only triggers if

`nextProps.location.search !== this.props.location.search`

we can probably completely return early here

Reply:
the comments in this function need to be moved around i think, particularly the one at the top

Reply:
We can't early return cos searchId might have changed